content/using-input-fields/README.md

```diff
@@ -0,0 +1,383 @@
+
+```js
+// import html helper
+var html = require('choo/html')
+
+// export function
+module.exports = function (state, emit) {
+  return html`
+    <div class="container">
+      <div class="grass">
+        <div class="lion">
+          <img src="/assets/lion.gif">
+          <div class="bubble"></div>
+        </div>
+
+        <div class="crocodile">
+          <img src="/assets/crocodile.gif">
+          <div class="bubble"></div>
+        </div>
+      </div>
+    </div>
+  `
+}
+```
+
+At the top of this file, we've imported the `html` function that comes with `choo`, which allows us to create HTML templates that `choo` can understand and render on the page. Below, we're exporting a function that returns a template which will render a square of grass, a lion, a crocodile and some speech bubbles (the assets and CSS styles for these HTML elements are included with the starter project).
+
+We can see what our app currently looks like, by clicking the "Show" button near the top left hand corner of the window:
+
+![starter1](images/starter1.gif "Screenshot of starter project")
+
+Let's add some input fields to our application. Modify the HTML template inside `template/main.js` to read the following:
+
+```html
+<div class="container">
+  <div class="grass">
+    <div class="lion">
+      <img src="/assets/lion.gif">
+      <div class="bubble"></div>
+    </div>
+
+    <div class="crocodile">
+      <img src="/assets/crocodile.gif">
+      <div class="bubble"></div>
+    </div>
+  </div>
+
+  <div class="inputs">
+    <div class="input">
+      <div class="label">The Lion says</div>
+      <input type="text" id="lion" />
+    </div>
+
+    <div class="input">
+      <div class="label">The Crocodile says</div>
+      <input type="text" id="crocodile" />
+      <button>Update</button>
+    </div>
+  </div>
+</div>
+```
+
+Let's see what our app looks like now:
+
+![inputs1](images/inputs1.gif "Screenshot of new input fields")
+
+Below our grass square (`<div class="grass">`), we've a added new container (`<div class="inputs">`) which includes two input fields (`<input type="text" />`), and a `<button>`. The first input field is for our Lion, and the second input field is for our Crocodile. At this point, if we type some text into one of these inputs, or click on the button, nothing will happen. However, we can add properties to these elements which will trigger a function when an interaction occurs.
+
+## Adding event handlers to inputs
+The idea of this application is that it should demonstrate two different ways in which we can get text from an input field to our application's state.
+
+One way we can do this is by detecting when the contents of an input field has changed. In response, we can pass this updated input value to our application's state.
+
+Let's update the code of our first input field:
+
+```html
+<div class="input">
+  <div class="label">The Lion says</div>
+  <input type="text" id="lion" oninput=${updateLion} />
```

Comment:
Not sure if relevant in this situation; but setting a default value from `state` might make sense - that way on re-renders the value isn't cleared. Not sure if it matters too much here tho

```html
<input value=${state.lion} />
```